print-bgp.c

```diff
@@ -2506,6 +2510,56 @@ bgp_attr_print(netdissect_options *ndo,
             len -= 12;
         }
         break;
+    case BGPTYPE_BGPSEC_PATH:
+    {
+        uint16_t sblen, splen, siglen;
+        u_int idx;
+
+        splen = GET_BE_U_2(tptr);
+        idx = 0;
+
+        if (splen < 8) {
+            ND_PRINT(" [secure path too short]");
+            tptr += splen;
+            break;
+        }
+
+        ND_PRINT("\n\t    Secure Path Length: %u", splen);
+
+        idx += 2;
+        splen -= 2;
+        /* parse secure path segments */
+        while (splen > 0) {
+            ND_PRINT("\n\t      Secure Path Segment: pCount: %u, Flags: %u, AS: %u",
+                     GET_U_1(tptr + idx),
+                     GET_U_1(tptr + idx + 1),
```

Comment:
Should this use `bittok2str()` or `bittok2str_nosep()` to show the individual flags (or flag, in this case), rather than just dumping the flag value as a decimal number?

Reply:
Right, `Flags` are currently ignored (and are thus always 0). I oriented myself on the example here:
https://datatracker.ietf.org/doc/html/rfc8208#appendix-A.4
There, the flags are also displayed. Whether or not this is useful, I am not sure myself.

Reply:
Just like with capabilities, here, only the leftmost bit of the octet is relevant (see [RFC 8205 Section 3.1](https://datatracker.ietf.org/doc/html/rfc8205#section-3.1)). To make the output consistent with the rest of the code, I would suggest to scrap the `Flags` field.

Reply:
One more question: since there is only one possible flag to be set (or not set), how to best display the value?
`Flags: [0/1]`
or
`Flags: [Set/Unset]`
or leave out the `Flags` field entirely if its value is 0?

Reply:
The specification calls the other 7 bits "unassigned" and explains they are a typical case of "must be zero", so you could spell the individual bits as `MBZ-n` (as done in `print-udld.c`, for example) and use `bittok2str()`.

Reply:
Thank you very much for the pointer. I updated the code and the example output accordingly. Let me know if is as you intended it to be.